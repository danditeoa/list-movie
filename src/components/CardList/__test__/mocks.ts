export const mockList = [
  {
    adult: false,
    backdrop_path: '/y5Z0WesTjvn59jP6yo459eUsbli.jpg',
    genre_ids: [27, 53],
    id: 663712,
    original_language: 'en',
    original_title: 'Terrifier 2',
    overview:
      "After being resurrected by a sinister entity, Art the Clown returns to Miles County where he must hunt down and destroy a teenage girl and her younger brother on Halloween night.  As the body count rises, the siblings fight to stay alive while uncovering the true nature of Art's evil intent.",
    popularity: 7866.782,
    poster_path: '/b6IRp6Pl2Fsq37r9jFhGoLtaqHm.jpg',
    release_date: '2022-10-06',
    title: 'Terrifier 2',
    video: false,
    vote_average: 7.1,
    vote_count: 445,
  },
  {
    adult: false,
    backdrop_path: '/bQXAqRx2Fgc46uCVWgoPz5L5Dtr.jpg',
    genre_ids: [28, 14, 878],
    id: 436270,
    original_language: 'en',
    original_title: 'Black Adam',
    overview:
      'Nearly 5,000 years after he was bestowed with the almighty powers of the Egyptian gods—and imprisoned just as quickly—Black Adam is freed from his earthly tomb, ready to unleash his unique form of justice on the modern world.',
    popularity: 4428.609,
    poster_path: '/3zXceNTtyj5FLjwQXuPvLYK5YYL.jpg',
    release_date: '2022-10-19',
    title: 'Black Adam',
    video: false,
    vote_average: 6.9,
    vote_count: 905,
  },
  {
    adult: false,
    backdrop_path: '/pGx6O6IwqADOsgmqWzPysmWnOyr.jpg',
    genre_ids: [28, 14],
    id: 732459,
    original_language: 'en',
    original_title: 'Blade of the 47 Ronin',
    overview:
      'In this sequel to "47 Ronin," a new class of warriors emerges among the Samurai clans to keep a sought-after sword from falling into the wrong hands.',
    popularity: 3821.739,
    poster_path: '/kjFDIlUCJkcpFxYKtE6OsGcAfQQ.jpg',
    release_date: '2022-10-25',
    title: 'Blade of the 47 Ronin',
    video: false,
    vote_average: 6.7,
    vote_count: 47,
  },
  {
    adult: false,
    backdrop_path: '/naNXYdBzTEb1KwOdi1RbBkM9Zv1.jpg',
    genre_ids: [27, 53],
    id: 420634,
    original_language: 'en',
    original_title: 'Terrifier',
    overview:
      'On Halloween night, a young woman finds herself as the obsession of a sadistic murderer known as Art the Clown.',
    popularity: 3031.419,
    poster_path: '/6PQqC4SbY910VvyVad6mvsboILU.jpg',
    release_date: '2016-10-15',
    title: 'Terrifier',
    video: false,
    vote_average: 6.6,
    vote_count: 889,
  },
  {
    adult: false,
    backdrop_path: '/b2FxWOxe9K7ZZ1uaPOze2RJ1ajq.jpg',
    genre_ids: [27, 28, 35],
    id: 675054,
    original_language: 'es',
    original_title: 'MexZombies',
    overview:
      'A group of teenagers must face a zombie apocalypse, and help reestablish order.',
    popularity: 2784.235,
    poster_path: '/pTxw4GFE3ZfzJfvuUsPyD5njAuI.jpg',
    release_date: '2022-10-26',
    title: 'MexZombies',
    video: false,
    vote_average: 7.3,
    vote_count: 49,
  },
  {
    adult: false,
    backdrop_path: '/1DBDwevWS8OhiT3wqqlW7KGPd6m.jpg',
    genre_ids: [53],
    id: 985939,
    original_language: 'en',
    original_title: 'Fall',
    overview:
      'For best friends Becky and Hunter, life is all about conquering fears and pushing limits. But after they climb 2,000 feet to the top of a remote, abandoned radio tower, they find themselves stranded with no way down. Now Becky and Hunter’s expert climbing skills will be put to the ultimate test as they desperately fight to survive the elements, a lack of supplies, and vertigo-inducing heights',
    popularity: 2350.808,
    poster_path: '/spCAxD99U1A6jsiePFoqdEcY0dG.jpg',
    release_date: '2022-08-11',
    title: 'Fall',
    video: false,
    vote_average: 7.3,
    vote_count: 1650,
  },
];
