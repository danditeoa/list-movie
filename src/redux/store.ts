import { configureStore, combineReducers } from "@reduxjs/toolkit";
import { popularMovies } from "./popular/popular-reducers";
import { searchResults } from "./search/search-reducers";

const rootReducer = combineReducers({ popularMovies, searchResults });

const store = configureStore({
  reducer: rootReducer,
  devTools: process.env.NODE_ENV === "development",
});

export default store