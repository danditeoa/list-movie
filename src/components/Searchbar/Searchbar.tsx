import React, { useState } from "react"
import { useDispatch } from "react-redux"
import { searchThunk } from "../../redux/search/search-thunk"
import { AppDispatch } from "../../redux/store"
import "./Searchbar.scss"

const Searchbar = () => {
  const [searchValue, setSearchValue] = useState("")

  const handleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
    setSearchValue(event.target.value)
  }

  const dispatch = useDispatch<AppDispatch>()

  return (
    <div className="searchbar-container">
      <input
        type="text"
        placeholder="search for a movie"
        value={searchValue}
        onChange={handleChange}
      />
      <button onClick={() => dispatch(searchThunk(searchValue))}>Search</button>
    </div>
  )
}

export default Searchbar
