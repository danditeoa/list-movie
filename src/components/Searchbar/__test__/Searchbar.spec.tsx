import { fireEvent, render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import * as ReactRedux from 'react-redux';
import store from '../../../redux/store';
import { mockList } from '../../CardList/__test__/mocks';
import Searchbar from '../Searchbar';

jest.mock('react-redux', () => ({
  ...jest.requireActual('react-redux'),
  useDispatch: () => jest.fn(() => Promise.resolve([mockList[0]])),
}));

// jest.mock('../../store/foo/actions', () => ({
//   myActionCreator: jest.fn(
//     () => () => Promise.resolve([{ id: 1, username: 'foo' }])
//   ),
// }));

const getSearchBar = () => (
  <ReactRedux.Provider store={store}>
    <Searchbar />
  </ReactRedux.Provider>
);

describe('<Searchbar />', () => {
  const useDispatchMock = jest.spyOn(ReactRedux, 'useDispatch');

  beforeEach(() => {
    useDispatchMock.mockClear();
  });

  it('matches snapshot', () => {
    const { container } = render(getSearchBar());
    expect(container).toMatchSnapshot();
  });

  it('returns film search', () => {
    const dummyDispatch = jest.fn();
    useDispatchMock.mockReturnValue(dummyDispatch);
    expect(dummyDispatch).not.toHaveBeenCalled();

    render(getSearchBar());

    userEvent.type(screen.getByRole('textbox'), 'Hello,{enter}World!');

    fireEvent.click(screen.getByText('Search'));

    expect(dummyDispatch).toHaveBeenCalledWith({
      payload: { results: {} },
      type: 'search/request',
    });
  });
});
