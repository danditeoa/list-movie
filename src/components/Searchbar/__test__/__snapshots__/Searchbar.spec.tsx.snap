// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Searchbar /> matches snapshot 1`] = `
<div>
  <div
    class="searchbar-container"
  >
    <input
      placeholder="search for a movie"
      type="text"
      value=""
    />
    <button>
      Search
    </button>
  </div>
</div>
`;
